### ISSUE TITLE
[SOLVED] Firefox has different audio backend when started with custom profile

### ISSUE THREAD

**Issue by omega3** [netblue30/firejail#4127]
Why when I start Firefox in Firejail like this:
`firejail --private=/home/user/jail/ /usr/lib/firefox/firefox`

And a new Firefox profile is created in  `/home/user/jail/.mozilla/firefox/`
I have in Firefox about:support → media: audio backend pulse-rust.

When I delete this Firefox profile from `/home/user/jail/.mozilla/firefox/`

And when I run it with custom profile:
`firejail --private=/home/user/jail/ --profile=/home/user/jail/.config/firejail/firefox.local /usr/lib/firefox/firefox`
new Firefox profile is created and when I run Firefox I have in about:support → media: audio backend alsa.

My config firefox.local contains only this

```
whitelist ${RUNUSER}/org.keepassxc.KeePassXC.BrowserServer
whitelist ${RUNUSER}/kpxc_server
```

It has nothing to do with sound.
Should I whitelist something that is related to pulse-rust?

Recently Manjaro switch to pipewire and it appeared that when there is alsa in about:support → media: audio backend in Firefox I can't hear sound. And it looks like the cause of the trouble is how Firefox profile is created with when Firefox is in Firejail with local settings.

**Environment**
Operating System: Manjaro Linux
KDE Plasma Version: 5.21.3
KDE Frameworks Version: 5.80.0
Qt Version: 5.15.2
Kernel Version: 5.11.6-1-MANJARO
OS Type: 64-bit
Graphics Platform: X11

```
firejail --version
firejail version 0.9.65

Compile time support:
        - AppArmor support is enabled
        - AppImage support is enabled
        - chroot support is enabled
        - D-BUS proxy support is enabled
        - file and directory whitelisting support is enabled
        - file transfer support is enabled
        - firetunnel support is enabled
        - networking support is enabled
        - overlayfs support is enabled
        - private-home support is enabled
        - private-cache and tmpfs as user enabled
        - SELinux support is disabled
        - user namespace support is enabled
        - X11 sandboxing support is enabled
```

**Comment by rusty-snake** [comment]
`--profile=/home/user/jail/.config/firejail/firefox.local` means you use only these two line which restrict access to `/run/user/$UID`. You have NO `seccomp`, `nonewprivs`, `caps.drop all`, `disable-mnt` or `dbus-user filter`. Are you sure you want this?

Regarding your question: If firefox can not access the pulseaudio/pipewire socket, it uses alsa as fallback.

**Comment by omega3** [comment]
> Are you sure you want this?
I don't even understand what it means. So far didn't I use local profile but there were changes in Keepassxc so I had to use it. 

> If firefox can not access the pulseaudio/pipewire socket,
So I need to add my local profile whitelist to Firefox profile or copy from Firefox profile to local profile all else?

**Comment by rusty-snake** [comment]
> looks a different from https:/github.com/netblue30/firejail/blob/master/etc-fixes/**0.9.52**/firefox.profile

You are at 0.9.65. &nbsp; :thinking: &nbsp; The git-master profile is at https://github.com/netblue30/firejail/blob/master/etc/profile-a-l/firefox.profile.

> Are profiles in /etc/firejail/ updated when Firejail is updated? 

Yes. apt, dnf, pacman update `/etc/firejail/*` if you update firejail/firejail-profiles.

>  I copied from /etc/firejail/firefox.profile

You can simply add these two whitelist (for kpxc) to `~/.config/firejail/firefox.local` and start you sandbox with `firejail --private=~/jail /usr/bin/firefox`. Furthermore you can even more the `private ${HOME}/jail` to `firefox.local`.

**Comment by omega3** [comment]
Ok. Thank you very much. It is solved. You may close it.
